### PR TITLE
feat: doc structures handle decorators

### PR DESCRIPTION
**Pull request by kitsonk**
Closes: #133

This also changes "<UNIMPLEMENTED>" to "[UNSUPPORTED]" across the board, as especially when rendering SSR, the unencoded `<` and `>` cause problems in display, and if the consumer doesn't sanitise it, you get strange results.

### CODE REVIEW

**Comment by kitsonk** [comment]
Also, side note, I don't think swc supports [parameter decorators](https://www.typescriptlang.org/docs/handbook/decorators.html#parameter-decorators) and so the seralized structures don't currently support it.  This PR supports class, method/function, and property decorators.

**Comment by andreespirela** [comment]
Is there a version set for this? Is there such thing as canary for deno doc? Thanks!

**Comment by kitsonk** [comment]
For it as part of the CLI, the PR needs to be merged and a version of the `deno_doc` crate needs to be published, and the CLI needs to update to that crate version.  That will all likely take place before 1.14.  To use the JavaScript/WASM interface, you could do that right now by:

```ts
import { doc } from "https://raw.githubusercontent.com/kitsonk/deno_doc/kitsonk/issue133/mod.ts";

console.log(await doc("https://example.com/mod.ts"));
```

Though the branch will get deleted once the PR is merged.

**Comment by andreespirela** [comment]
@kitsonk can you generate JSON with the wasm interface? Any guidence towards that?

**Comment by kitsonk** [comment]
@andreespirela https://github.com/denoland/deno_doc#doc

**Comment by andreespirela** [comment]
@kitsonk the API doesn’t accept import maps right? If not, do they take the main app import map into consideration?

**Comment by kitsonk** [comment]
@andreespirela it does, in the sense of a resolver, but there isn't a default implementation of a resolver that takes an import map at the moment.

File: src/class.rs
Comment (bartlomieju) [inline src/class.rs:70]: Does it make sense to skip serializing if vector is empty? We don't do that for any other elements

File: src/decorators.rs
Comment (bartlomieju) [inline src/decorators.rs:21]: Ditto

**Review by bartlomieju — APPROVED** [review]
LGTM, but I'm unsure if we should skip serializing the field if there's no decorator

File: src/class.rs
Comment (kitsonk) [reply src/class.rs:70]: I want to migrate to it in the future, and since this is additive it isn't a breaking change.  Having loads of properties on objects with empty arrays has a non-negligible impact on performance, when you start doing that by thousands ands thousands of nodes that can be in an JSON structure.  Any empty array adds no information and just causes "bloat".  We don't print out `there are no methods` in we format it for display, so it doesn't increase readability or usability.

Comment (bartlomieju) [reply src/class.rs:70]: Okay, that's fair.

**Pull request merged**
